ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -389,25 +354,17 @@ private boolean updateBlobMetadata(BlobId blobId, String fieldName, Object value
         storageTimer.stop();
       }
 
-      Timer.Context docTimer = azureMetrics.documentUpdateTime.time();
-      try {
-        String docLink = cosmosCollectionLink + "/docs/" + blobId.getID();
-        RequestOptions options = new RequestOptions();
-        options.setPartitionKey(new PartitionKey(blobId.getPartition().toPathString()));
-        ResourceResponse<Document> response = documentClient.readDocument(docLink, options);
-        //CloudBlobMetadata blobMetadata = response.getResource().toObject(CloudBlobMetadata.class);
-        Document doc = response.getResource();
-        if (doc == null) {
-          logger.warn("Blob metadata record not found: " + docLink);
-          return false;
-        }
-        // Update only if value has changed
-        if (!value.equals(doc.get(fieldName))) {
-          doc.set(fieldName, value);
-          documentClient.replaceDocument(doc, options);
-        }
-      } finally {
-        docTimer.stop();
+      ResourceResponse<Document> response = cosmosDataAccessor.readMetadata(blobId);
+      //CloudBlobMetadata blobMetadata = response.getResource().toObject(CloudBlobMetadata.class);
+      Document doc = response.getResource();
+      if (doc == null) {
+        logger.warn("Blob metadata record not found: " + blobId.getID());
```

Comment:
Use {}?
```logger.warn("Blob metadata record not found: {}", blobId.getID());```

Reply:
I'll change it.  In general, this only makes a difference for debug statements.  Warnings are always logged so string joining will happen regardless.